src/app.js

```diff
@@ -96,39 +81,62 @@ const posts = [
   }
 ];
 
+Object.assign(app.locals, {
+  meta: {
+    title: 'My Blog',
+    description: 'A blog about something awesome!'
+  },
+  header: {
+    title: 'Something Awesome'
+  },
+  footer: {
+    year: new Date().getFullYear()
+  },
+  pages: pages
+});
+
+app.use(middleware.currentPath);
+
 app.get('/', function(req, res) {
 
   res.render('index', {
-    posts: posts,
+    posts: posts
   });
 
 });
 
-app.get('/about', function(req, res) {
-
-  res.render('page', {
-    page: pages.find(function (page){
-      return page.id === 'about';
-    })
-  })
-});
+app.get('/posts/:id', function(req, res) {
+  const post = posts.find(function (post) {
+    return post.id === Number(req.params.id)
+  });
 
-app.get('/contact', function(req, res) {
+  if (post) {res.render('post', {
```

Comment:
This should be two lines (hit enter after the `{`), and you should `return res.render(...);` to avoid calling `res.sendStatus(404);` afterwards. The `return` statement will prevent any more code from being executed in this function; without it, the program will continue executing commands after the `if` statement.